### PR TITLE
ML.NET: New landing page & TOC

### PR DESCRIPTION
**Pull request by cjgronlund**
Per product team request: New landing page and TOC for ML.NET

Preview link: https://review.docs.microsoft.com/en-us/dotnet/machine-learning/index?branch=pr-en-us-11855

Separate ML.NET TOC from large .NET TOC
Create landing page based on Linux VM landing page -- per agreement with Maira. :>)

@natke

### CODE REVIEW

File: docs/toc.yml
Comment (mairaw) [inline docs/toc.yml]: don't you have to delete the index.md file in this case?

**Comment by cjgronlund** [comment]
@mairaw: Are aware of an issue with YML landing pages? If not, I'll go to Docs Support to help with issues with the build.

**Comment by mairaw** [comment]
I’ve never done it so I don’t know 🤷‍♀️ 
@billwagner has some experience with yml files in this repo

**Comment by cjgronlund** [comment]
Thanks, @mairaw 
@BillWagner: I think I found the issues -- looks like this might have been user error. Fingers crossed.

File: docs/toc.yml
Comment (cjgronlund) [reply docs/toc.yml]: @mairaw Please review the updated version and let me know if this is still an issue. Thanks! :)

**Comment by cjgronlund** [comment]
@mairaw @natke 
Please review. Build issues and broken links in in the warnings were resolved.

**Review by mairaw — APPROVED** [review]
LGTM @cjgronlund! When you're ready to merge, remove the WIP from the title.

**Review by natke — CHANGES_REQUESTED** [review]
Something strange is happening with this doc: https://review.docs.microsoft.com/en-us/dotnet/machine-learning/basic-concepts-model-training-in-mldotnet?branch=pr-en-us-11855. It goes back to the old TOC. Also, the styling is not quite the same as the old docset (blue highlight over selected document in the TOC. Do we care?)

> #resolved This was caused by an incremental build issue

**Comment by mairaw** [comment]
The style is probably because the TOC preview feature is enabled on review for the upcoming changes.

**Comment by mairaw** [comment]
But I have no clue why the old TOC would be loading for that entry. Good catch @natke!

**Comment by cjgronlund** [comment]
@natke - your call. Do you want this live now? Thank you, @mairaw!

**Comment by cjgronlund** [comment]
@natke 
I kicked the side of the soda machine and the build is OK now. Merge? :)
Good catch, BTW!

**Review by natke — APPROVED** [review]

**Review by JRAlexander — APPROVED** [review]
LGTM. :shipit: when ready!

**Pull request merged**
